app/src/main/java/com/claraberriel/mvpparking/mvp/view/ReservationView.java

```diff
@@ -0,0 +1,119 @@
+    private final Context context;
+    private final Picker startDatePicker;
+    private final Picker endDatePicker;
+    
+    public ReservationView(Fragment fragmentRef, FragmentReservationBinding reservationBinding) {
+        super(fragmentRef);
+        this.reservationBinding = reservationBinding;
+        this.context = getContext();
+        startDatePicker = new Picker(reservationBinding.startDateTime);
+        endDatePicker = new Picker(reservationBinding.endDateTime);
+    }
+
+    /**
+     * Show Date and Time Dialogs
+     */
+
+    public void startDateTimeDialog() {
+        startDatePicker.show();
+    }
+
+    public void endDateTimeDialog() {
+        endDatePicker.show();
+    }
+
+    /**
+     * Getters
+     * @return values from EditTexts inputs
+     */
+
+    public Date getStartDate() {
+        return startDatePicker != null ? startDatePicker.getDate() : null;
+    }
+
+    public Date getEndDate(){
+        return endDatePicker != null ? endDatePicker.getDate() : null;
+    }
+
+    public String getParkingNumber() throws NumberFormatException {
+            return reservationBinding.parkingNumber.getText().toString();
+    }
+
+    public String getSecurityCode() {
+        return reservationBinding.securityCode.getText().toString();
+    }
+
+    /**
+     * Toast Messages
+     * @param message to be displayed
+     */
+
+    //reusable
+    private void showToast(String message) {
+        Toast.makeText(context, message, Toast.LENGTH_LONG).show();
+    }
+
+    public void showErrorNoReservationInThePast() {
+        if (context != null){
+            showToast((context).getString(R.string.reservationview_err_msg_inthepast));
+        }
+    }
+
+    public void showMissingField() {
+        if (context != null){
+            showToast((context).getString(R.string.reservationview_err_msg_missingfield));
+        }
+    }
+
+    public void showBackToTheFuture() {
+        if (context !=null){
+            showToast((context).getString(R.string.reservationview_err_msg_delorean));
+        }
+    }
+
+    public void showInvalidNumber() {
+        if (context != null){
+            showToast((context).getString(R.string.reservationview_err_msg_numberformat));
+        }
+    }
+
+    public void showZeroNotAccepted() {
+        if (context != null) {
+            showToast((context).getString(R.string.reservationview_err_msg_zero));
+        }
+    }
+
+    public void showLargeNumber(int value) {
+        if (context != null) {
+            showToast((context).getString(R.string.reservationview_err_msg_large, value));
+        }
+    }
+
+    public void showLargerThanThree() {
+        if (context != null){
+            showToast((context).getString(R.string.reservationview_err_msg_largerthan3));
+        }
+    }
+    
+    public void showReservationSuccess() {
+        if (context != null){
+            showToast((context).getString(R.string.reservationview_msg_success));
+        }
+    }
```

Comment:
Refactor this.

Reply:
Do you mean refactoring the method names? I just changes the context to getContext() and removed the ifs by applying it only on the parent method showToast()